toolbox/src/DiscreteFilter.cpp

```diff
@@ -125,7 +132,10 @@ bool DiscreteFilter::initialize(BlockInformation* blockInfo, wbt::Error* error)
             }
             return 1;
         }
-        filter = new iCub::ctrl::Filter(*num, *den);
+        for (unsigned i = 0; i < inputSignalWidth; ++i) {
```

Comment:
With the new commit I forced pushed now the block works with vectors and reduces dynamic allocation